core/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -206,12 +203,13 @@ public DocumentMapper(MapperService mapperService, @Nullable Settings indexSetti
         MapperUtils.collect(this.mapping.root, newObjectMappers, newFieldMappers);
 
         this.fieldMappers = new DocumentFieldMappers(docMapperParser.analysisService).copyAndAllAll(newFieldMappers);
-        this.objectMappers = Maps.uniqueIndex(newObjectMappers, new Function<ObjectMapper, String>() {
-            @Override
-            public String apply(ObjectMapper mapper) {
-                return mapper.fullPath();
-            }
-        });
+
+        Map<String, ObjectMapper> builder = new HashMap<>();
+        for (ObjectMapper objectMapper : newObjectMappers) {
+            builder.put(objectMapper.fullPath(), objectMapper);
+        }
+
+        this.objectMappers = Collections.unmodifiableMap(builder);
```

Comment:
Just to ensure that there were no key collisions? What do you think of 43cc26ef4092fbef62060e16b9c540777fe902fc?